### PR TITLE
Expose ChangeNotifier.isDisposed

### LINKED ISSUE #71983: [production] ChangeNotifier.removeListener: Null check operator used on a null value
I see a lot of these exceptions in Crashlytics for my app:
```bash
Non-fatal Exception: io.flutter.plugins.firebase.crashlytics.FlutterError: Null check operator used on a null value. Error thrown Instance of 'ErrorSummary'.
       at ChangeNotifier.removeListener(ChangeNotifier.java:179)
       at EditableTextState.dispose(EditableTextState.java:1590)
       at StatefulElement.unmount(StatefulElement.java:4854)
       at _InactiveElements._unmount(_InactiveElements.java:1998)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at Iterable.forEach(Iterable.java)
       at _RenderChipElement.visitChildren(_RenderChipElement.java:2079)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at MultiChildRenderObjectElement.visitChildren(MultiChildRenderObjectElement.java:6218)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at MultiChildRenderObjectElement.visitChildren(MultiChildRenderObjectElement.java:6218)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at MultiChildRenderObjectElement.visitChildren(MultiChildRenderObjectElement.java:6218)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at _SliverPersistentHeaderElement.visitChildren(_SliverPersistentHeaderElement.java:270)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at _InactiveElements.<fn>(_InactiveElements.java:1996)
       at ComponentElement.visitChildren(ComponentElement.java:4681)
       at _InactiveElements._unmount(_InactiveElements.java:1994)
       at ListIterable.forEach(ListIterable.java)
       at _InactiveElements._unmountAll(_InactiveElements.java:2007)
       at BuildOwner.<fn>(BuildOwner.java:2821)
       at BaseMouseTracker._deviceUpdatePhase(BaseMouseTracker.java:317)
       at BuildOwner.finalizeTree(BuildOwner.java:2820)
       at WidgetsBinding.drawFrame(WidgetsBinding.java:915)
       at RendererBinding._handlePersistentFrameCallback(RendererBinding.java:302)
       at SchedulerBinding._invokeFrameCallback(SchedulerBinding.java:1117)
       at SchedulerBinding.handleDrawFrame(SchedulerBinding.java:1055)
       at SchedulerBinding._handleDrawFrame(SchedulerBinding.java:971)
```
Stack trace is from Android, but there is same issue in iOS. I have no idea where in my code this exception happens.
```bash
[√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.19041.685], locale en-US)
    • Flutter version 1.22.4 at C:\tools\fluttersdk\flutter
    • Framework revision 1aafb3a8b9 (4 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at c:\tools\android-sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = C:\tools\android-sdk
    • ANDROID_SDK_ROOT = C:\tools\android-sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.51.1)
    • VS Code at C:\Users\me\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.17.0

[√] Connected device (1 available)
    • SM G930F (mobile) • ad0c16039914a951e1 • android-arm64 • Android 8.0.0 (API 26)
```

### PR DESCRIPTION
**Pull request by caseycrogers**
In many situations, it is impossible to safely call `removeListener` from a `StatefulWidget`'s `dispose` function. In many cases the stateful widget, or even the developer's app, is not in control of the change notifier and the widget may be disposed before or after the change notifier was disposed. For example, the change notifier may be exposed by a third party package that makes no guarantees about the order in which a change notifier will be disposed. See #71983 for a specific example.

Ideally, calls to `removeListener` should be safe-it shouldn't throw when the change notifier has already been disposed. However, removing the assert is theoretically a breaking change as some users may have `try...on` blocks that rely on the assert throwing in this case. To avoid a breaking change, I've instead exposed `_debugIsDisposed` which is more broadly useful than just for ensuring calls to `removeListener` are safe.

It appears that `ChangeNotifier` has been written with a philosophical belief that `_debugIsDisposed` should not be public. However, you can't have both. Either expose `isDisposed`, or make `removeListener` safe to call. Otherwise you're forcing developers in some circumstances to use sloppy and dangerous `try...on` blocks. Honestly, ideally, we'd want to expose `isDisposed` AND make calls to `removeListener` safe.

Fixes: https://github.com/flutter/flutter/issues/71983

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by caseycrogers** [comment]
@goderbauer Hey! I've created this PR to resolve #71983. I noticed you've reviewed some previous PRs against `ChangeNotifier`-could you look at this one for me?

File: packages/flutter/lib/src/foundation/change_notifier.dart [hunk 1]
```diff
@@ -106,11 +106,17 @@ class ChangeNotifier implements Listenable {
   List<VoidCallback?> _listeners = List<VoidCallback?>.filled(0, null);
   int _notificationCallStackDepth = 0;
   int _reentrantlyRemovedListeners = 0;
-  bool _debugDisposed = false;
+  bool _isDisposed = false;
```
Comment (caseycrogers) [inline packages/flutter/lib/src/foundation/change_notifier.dart:109]: Given that this field isn't strictly for debugging (it's now indirectly exposed to developers), I've renamed it and it's associated assert function.

**Comment by goderbauer** [comment]
Looks like the analyzer is unhappy with this change and it is breaking some customers, which makes this a breaking change. Those require special treatment: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Regardless of that, I am not in favor of actually making this change. If you're trying to operate on a disposed ChangeNotifier that usually indicates that there's a lifecycle bug somewhere.

**Comment by caseycrogers** [comment]
> Looks like the analyzer is unhappy with this change and it is breaking some customers, which makes this a breaking change. Those require special treatment: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
> 
> Regardless of that, I am not in favor of actually making this change. If you're trying to operate on a disposed ChangeNotifier that usually indicates that there's a lifecycle bug somewhere.

Oof yeah I didn't think it through carefully enough-anyone using `implements` instead of `extends` will be broken by this-very non-ideal.

This makes making calls to `removeListener` safe a much more appealing solution.

If you don't think this change is desirable, can you post in the associated bug why? The issue being brought up there is the developer is often not in control of the notifier's lifecycle. eg if they're getting it from a third party package. In those cases, they may not know if they notifier or their widget will be disposed of first and, even worse, there may be race conditions that make the dispose order non-deterministic.

**Comment by slavap** [comment]
> Regardless of that, I am not in favor of actually making this change. If you're trying to operate on a disposed ChangeNotifier that usually indicates that there's a lifecycle bug somewhere.

The keyword here is "somewhere" - you cannot do anything meaningful if third party code involved, the only way is to use try..catch for removeListener

**Comment by goderbauer** [comment]
> The issue being brought up there is the developer is often not in control of the notifier's lifecycle. eg if they're getting it from a third party package. In those cases, they may not know if the notifier or their widget will be disposed of first and, even worse, there may be race conditions that make the dispose order non-deterministic.

How can you meaningfully use the ChangeNotifier of a third party packages if said package is just willy-nilly disposing the notifier? There must be some kind of a lifecycle contract - and this contract should involve giving the consumer a change to properly release its usage of the notifier before it is disposed.

This change (or the proposed change to removeListener) is just papering over another bug in the system which may then blow up in a different place.

**Comment by slavap** [comment]
@goderbauer 
>> This change (or the proposed change to removeListener) is just papering over another bug in the system which may then blow up in a different place.

First of all - removeListener must never raise any exceptions - nobody cares if it is already disposed or not. If nothing to do then do nothing without raising useless exceptions.

Second - there is reproducible case, take a look: https://github.com/flutter/flutter/issues/71983#issuecomment-917297746

**Comment by caseycrogers** [comment]
> @goderbauer
> 
> > > This change (or the proposed change to removeListener) is just papering over another bug in the system which may then blow up in a different place.
> 
> First of all - removeListener must never raise any exceptions - nobody cares if it is already disposed or not. If nothing to do then do nothing without raising useless exceptions.
> 
> Second - there is reproducible case, take a look: [#71983 (comment)](https://github.com/flutter/flutter/issues/71983#issuecomment-917297746)

FWIW, he just resolved my related issue-I was misunderstanding the intended use of `Scrollable.of` and how route interacts with it:
https://github.com/flutter/flutter/issues/89924#issuecomment-920424050

I still agree on `#1`. Cleanup functions should only throw in fairly exceptional circumstances-it's clear this philosophy was already in mind when it was decided that `removeListener` should not throw if the specified function is not present. The same logic that says it shouldn't throw if the function isn't there is just as compelling for saying it shouldn't throw if the notifier is already disposed.

**Pull request closed without merging**